internal/controller/daemonset/instaslice_daemonset.go

```diff
@@ -126,8 +127,26 @@ func (r *InstaSliceDaemonsetReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	var instaslice inferencev1alpha1.Instaslice
 	if err := r.Get(ctx, nsName, &instaslice); err != nil {
+		if apierrors.IsNotFound(err) {
+			var node v1.Node
+			if err := r.Get(ctx, client.ObjectKey{Name: r.NodeName}, &node); err != nil {
+				log.Error(err, "error fetching node for missing Instaslice")
+				return ctrl.Result{RequeueAfter: controller.Requeue1sDelay}, nil
+			}
+			if _, ok := node.Labels[controller.ManagedLabel]; !ok {
+				log.Info("Node is not labeled managed, skipping Instaslice discovery", "node", r.NodeName)
+				return ctrl.Result{RequeueAfter: controller.Requeue2sDelay}, nil
```

Comment:
If the node isn't label managed, then it shouldn't be RequeueAfter?

Reply:
Even if the node is unmanaged at a point, admins may manually label the node later. Without requeuing, the DaemonSet will never detect that change unless a new reconcile is triggered by another event.